addons/survey/tests/test_survey_results.py

```diff
@@ -0,0 +1,113 @@
+    The filters can be combined but their query count doesn't add up if their
+    related question answers data are stored in the same model.
+    """
+
+    def setUp(self):
+        super(TestSurveyResults, self).setUp()
+        self.SurveyController = Survey()
+
+    def test_get_filters_from_post(self):
+        """ Check that the filters are correctly retrieved from the post. """
+        post = {'filters': 'A,14,101|A,0,58|L,0,2'}
+        with MockRequest(self.env):
+            answer_by_column, user_input_lines_ids = self.SurveyController._get_filters_from_post(post)
+        self.assertEqual(answer_by_column, {101: [14], 58: []})
+        self.assertEqual(user_input_lines_ids, [2])
+
+        post = {'filters': 'A,0,9|J,40,3'}
+        # No model associated with the J letter, the second filter should be ignored
+        with MockRequest(self.env):
+            answer_by_column, user_input_lines_ids = self.SurveyController._get_filters_from_post(post)
+        self.assertEqual(answer_by_column, {9: []})
+        self.assertFalse(user_input_lines_ids)
+
+    def test_results_page_filters_survey_question_answer_model(self):
+        """ Check results and performance of matching one filter using
+        the 'survey.question.answer' answer model
+        """
+        post = {'filters': f'A,0,{self.gras_id}'}
+        expected_user_input_lines = self.user_input_1.user_input_line_ids + self.user_input_2.user_input_line_ids
+        self._check_results_and_query_count(post, expected_user_input_lines, 3)
+
+    def test_results_page_filters_survey_question_answer_model_multiple(self):
+        """ Check results and performance of matching multiple filters using
+        the 'survey.question.answer' answer model
+        """
+        post = {'filters': f'A,0,{self.gras_id}|A,0,{self.cat_id}'}
+        expected_user_input_lines = self.user_input_1.user_input_line_ids
+        self._check_results_and_query_count(post, expected_user_input_lines, 3)
+
+    def test_results_page_filters_survey_user_input_line_model(self):
+        """ Check results and performance of matching one filter using
+        the 'survey.user_input.line' answer model
+        """
+        post = {'filters': f'L,0,{self.answer_24.id}'}
+        expected_user_input_lines = self.user_input_1.user_input_line_ids + self.user_input_2.user_input_line_ids
+        self._check_results_and_query_count(post, expected_user_input_lines, 3)
+
+    def test_results_page_filters_survey_user_input_line_model_multiple(self):
+        """ Check results and performance of matching multiple filters using
+        the 'survey.user_input.line' answer model
+        """
+        post = {'filters': f'L,0,{self.answer_24.id}|L,0,{self.answer_pauline.id}'}
+        expected_user_input_lines = self.user_input_2.user_input_line_ids
+        self._check_results_and_query_count(post, expected_user_input_lines, 3)
+
+    def test_results_page_filters_survey_mixed_models(self):
```

Comment:
We usually prefer when test methods are sorted alphabetically. (Utility methods stay at the bottom though, but those are also sorted alphabetically, here you only have one so it does not matter ^^).